src/features/contacts/index.tsx

```diff
@@ -9,51 +9,108 @@ import {
+  const [whereField, setWhereField] = useState("");
+  const [whereFieldValue, setWhereFieldValue] = useState("");
+  const [skipLimit, setSkipLimit] = useState(0);
+  const [totalRowCount, setTotalRowCount] = useState(0);
+
+  const contactsTableProps = {
+    contacts,
+    setPageSize: setFilterLimit,
+    pageSize: filterLimit,
+    setSkipLimit,
+    totalRowCount,
+    setSortColumn: setFilterOrderColumn,
+    setSortOrder: setFilterOrderDirection,
+    setFilterField: setWhereField,
+    setFilterFieldValue: setWhereFieldValue,
+  };
+
+  const basicFilters: FilterParams = {
+    "filter[limit]": filterLimit,
+    "filter[order]": `${filterOrderColumn} ${filterOrderDirection}`,
+    "filter[skip]": skipLimit,
+  };
+
+  const whereFilterQuery: string = whereFieldValue
+    ? `&filter[where][${whereField}]=${whereFieldValue}`
+    : "";
+
+  const basicFilterQuery = Object.keys(basicFilters)
+    .filter((key) => `${basicFilters[key].toString().trim()}` != "")
+    .map((key) => `${key}=${basicFilters[key]}`)
+    .join("&");
+
+  const setTotalResultsCount = (headerCount: string | null) => {
+    if (headerCount) setTotalRowCount(parseInt(headerCount, 10));
+    else setTotalRowCount(-1);
+  };
 
   useEffect(() => {
     (async () => {
-      try {
-        const { data } = await client.api.contactsList();
-        setContacts(data);
-      } catch (e) {
-        console.log(e);
-      }
+      const { data, headers } = await client.api.contactsList({
+        query: `${searchTerm}&${basicFilterQuery}${whereFilterQuery}`,
+      });
+      setTotalResultsCount(headers.get("x-total-count"));
+      setContacts(data);
     })();
-  }, [client]);
-
-  return (
-    <ModuleContainer>
-      <ModuleHeaderContainer>
-        <ModuleHeaderTitleContainer>
-          <Typography variant="h3">Contacts</Typography>
-        </ModuleHeaderTitleContainer>
-        <ModuleHeaderSubtitleContainer>
-          <Breadcrumbs separator={<NavigateNext fontSize="small" />}>
-            <Link to={rootRoute} component={GhostLink} underline="hover">
-              Dashboard
-            </Link>
-            <Typography variant="body1">Contacts</Typography>
-          </Breadcrumbs>
-        </ModuleHeaderSubtitleContainer>
-        <ModuleHeaderActionContainer>
-          <Button variant="contained">Add contact</Button>
-        </ModuleHeaderActionContainer>
-      </ModuleHeaderContainer>
-      <ExtraActionsContainer>
-        <Button startIcon={<Upload />}>Import</Button>
-        <Button startIcon={<Download />}>Export</Button>
-      </ExtraActionsContainer>
-      <ContactsTable contacts={contacts} />
-    </ModuleContainer>
-  );
+  }, [
+    searchTerm,
+    filterLimit,
+    skipLimit,
+    filterOrderColumn,
+    filterOrderDirection,
+    whereFieldValue,
+  ]);
+
+  if (totalRowCount === -1) {
+    throw new Error("Server error: x-total-count header is not provided.");
+  } else
```

Comment:
This `if` statement looks strange to me.. maybe it would be better to move this piece of logic to `useEffect` hook?
```
useEffect(() => {
  if (totalRowCount === -1) {
    throw new Error(...)
  }
}, [totalRowCount])

Reply:
Yes. Updated accordingly.